tinygrad/codegen/uops.py

```diff
@@ -307,30 +315,22 @@ def print(self):
     for i,u in enumerate(self):
       print(f"{i:4d} {str(u.op):20s}: {str(u.dtype) if u.dtype is not None else '':25s} " f"{str([self.uops.index(x) for x in u.src]):32s} {u.arg}")
 
-  def graph_rewrite(self, sink:UOp, pm:PatternMatcher):
-    # recursive rewrite
-    changed = getenv("UOPS_REWRITE", 1)
+  @functools.lru_cache(None)  # pylint: disable=method-cache-max-size-none
```

Comment:
Again, things now live forever.

Reply:
True. I see the risk - they can, however, be explicitly cleared after the graph rewrite (or after `linearize`). I added it [here](https://github.com/tinygrad/tinygrad/pull/5152/commits/4a998e665282e6dc556311cec84b0444c2b1aec5).

Reply:
Complexity!